modules/auxiliary/scanner/winrm/winrm_cmd.rb

```diff
@@ -31,17 +28,48 @@ def initialize
       ])
   end
 
-
   def run_host(ip)
-    streams = winrm_run_cmd(datastore['CMD'])
-    return unless streams.class == Hash
-    print_error streams['stderr'] unless streams['stderr'] == ''
-    print_good "#{peer}: #{streams['stdout']}"
-    path = store_loot("winrm.cmd_results", "text/plain", ip, streams['stdout'], "winrm_cmd_results.txt", "WinRM CMD Results")
-    print_good "Results saved to #{path}"
-  end
-
-
+    rhost = datastore['RHOST']
+    rport = datastore['RPORT']
+    uri = datastore['URI']
+    ssl = datastore['SSL']
+    schema = ssl ? 'https' : 'http'
+    endpoint = "#{schema}://#{rhost}:#{rport}#{uri}"
+    conn = Net::MsfWinRM::RexWinRMConnection.new(
+                endpoint: endpoint,
+                host: rhost,
+                port: rport,
+                uri: uri,
+                ssl: ssl,
+                user: datastore['USERNAME'],
+                password: datastore['PASSWORD'],
+                transport: :rexhttp,
+                :no_ssl_peer_verification => true,
+                :operation_timeout => 0.5,
+                :timeout => 20,
+                :retry_limit => 1,
+                :realm => datastore['DOMAIN']
+            )
```

Comment:
I think for Ruby 3 support we'd have to wrap this in an explicit hash
https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

```suggestion
    conn = Net::MsfWinRM::RexWinRMConnection.new(
           {
                endpoint: endpoint,
                host: rhost,
                port: rport,
                uri: uri,
                ssl: ssl,
                user: datastore['USERNAME'],
                password: datastore['PASSWORD'],
                transport: :rexhttp,
                :no_ssl_peer_verification => true,
                :operation_timeout => 0.5,
                :timeout => 20,
                :retry_limit => 1,
                :realm => datastore['DOMAIN']
            }
     )
```

Reply:
Change made - https://github.com/smashery/metasploit-framework/blob/winrm_improvements/modules/auxiliary/scanner/winrm/winrm_cmd.rb#L40-L54